oe_theme.theme

```diff
@@ -521,120 +521,163 @@ function oe_theme_preprocess_field(&$variables, $hook): void {
+  if ($pager_last > $pager_max) {
+    $pager_last = $pager_max;
   }
 
-  // Generate previous link if it is available.
-  if (!empty($variables['items']['previous']['href'])) {
-    $previous_link = [];
-    $previous_link['type'] = 'previous';
-    $previous_link['aria_label'] = t('Go to previous page');
-    $label = t('‹ Previous');
-    if (isset($tags[1])) {
-      $label = $tags[1];
-    }
-    $previous_link['link']['link'] = [
-      'path' => $variables['items']['previous']['href'],
-      'label' => $label,
+  // Array to collect the pagination items.
+  $ecl_items = [];
+
+  // Prepare navigation link item for previous/next links with defaults.
+  $nav_link = [
+    'link' => [
+      'icon' => [
+        'path' => $variables['ecl_icon_path'],
+        'type' => 'ui',
+        'name' => 'corner-arrow',
+        'size' => 'xs',
+        'transform' => 'rotate-270',
+      ],
+    ],
+  ];
+  // Prepare ellipsis item.
+  $ellipsis = [
+    'type' => 'ellipsis',
+    'label' => '...',
+  ];
+
+  // Add 'Previous' navigation link.
+  if ($pager_current > 1) {
+    $options = [
+      'query' => pager_query_add_page($parameters, $element, $pager_page_array[$element] - 1),
+    ];
+    $nav_link['link']['link'] = [
+      'path' => Url::fromRoute($route_name, $route_parameters, $options),
+      'label' => t('Previous'),
+      'icon_position' => 'before',
     ];
-    $variables['ecl_items'][] = $previous_link;
+    $nav_link['type'] = 'previous';
+    $nav_link['aria_label'] = t('Go to previous page');
+    $ecl_items[] = $nav_link;
   }
 
-  // Generate previous ellipses if needed.
-  if (isset($variables['ellipses']['previous']) && $variables['ellipses']['previous']) {
-    $ellipsis = [
-      'type' => 'ellipsis',
-      'label' => '...',
+  // Add first page link.
+  if ($pager_first != $pager_current) {
+    $options = [
+      'query' => pager_query_add_page($parameters, $element, 0),
+    ];
+    $ecl_items[] = [
+      'type' => 'first',
+      'link' => [
+        'link' => [
+          'path' => Url::fromRoute($route_name, $route_parameters, $options),
+          'label' => '1',
+        ],
+      ],
+      'aria_label' => t('Go to page @number', ['@number' => 1]),
     ];
-    $variables['ecl_items'][] = $ellipsis;
   }
 
-  // Generate actual pager items.
-  if (!empty($variables['items']['pages'])) {
-    foreach ($variables['items']['pages'] as $page_key => $page_item) {
-      if ($page_key == $variables['current']) {
-        $current_link = [
-          'type' => 'current',
-          'aria_label' => t('Page @number', ['@number' => $page_key]),
-          'label' => (string) $page_key,
-        ];
-        $variables['ecl_items'][] = $current_link;
+  // Add first ellipsis.
+  if ($pager_current - $pager_offset > 2) {
+    $ecl_items[] = $ellipsis;
+  }
+
+  // Generate page link items.
+  $i = $pager_first;
+  for (; $i <= $pager_current + $pager_offset && $i <= $pager_last; $i++) {
```

Comment:
Definition of index shouldn't be part of the for loop arguments?
We are not using $i outside of the for loop.